Add recipe for typst-preview.el

havarddj opened the pull request:
### Brief summary of what the package does

This package provides live editing of typst (https://github.com/typst) files, similar to impatient-mode. 

### Direct link to the package repository

https://github.com/havarddj/typst-preview.el

### Your association with the package

I am the creator and maintainer of this package.

### Relevant communications with the upstream package maintainer

**None needed** 

### Checklist


- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this. Quick first pass -

⸺ `typst-preview.el` with byte-compile using Emacs 30.1:
```
typst-preview.el:419:16: Warning: Unused lexical variable `browse-url-generic-program'
```

⸺ `typst-preview.el` with package-lint 20250828.1506:
```
17 issues found:
8:37: warning: Use a properly versioned dependency on "websocket" if possible.
8:47: warning: Use a properly versioned dependency on "f" if possible.
141:0: error: "tp--active-buffers" doesn't start with package's prefix "typst-preview".
145:0: error: "tp--master" doesn't start with package's prefix "typst-preview".
148:0: error: "tp--active-masters" doesn't start with package's prefix "typst-preview".
188:0: error: "tp--master-file" doesn't start with package's prefix "typst-preview".
194:0: error: "tp--local-master" doesn't start with package's prefix "typst-preview".
235:11: error: You should depend on (emacs "28.1") if you need `split-string-shell-command'.
378:0: error: "tp--parse-message" doesn't start with package's prefix "typst-preview".
391:0: error: "tp--editor-scroll-to" doesn't start with package's prefix "typst-preview".
410:0: error: "tp--connect-browser" doesn't start with package's prefix "typst-preview".
424:0: error: "tp--find-server-filter" doesn't start with package's prefix "typst-preview".
431:0: error: "tp--get-buffer-position" doesn't start with package's prefix "typst-preview".
436:0: error: "tp--goto-file-position" doesn't start with package's prefix "typst-preview".
449:0: error: "tp--stringify-buffer" doesn't start with package's prefix "typst-preview".
457:0: error: "tp--send-buffer" doesn't start with package's prefix "typst-preview".
467:0: error: "tp--send-buffer-on-type" doesn't start with package's prefix "typst-preview".
```

⸺ `typst-preview.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- typst-preview.el#L245: It's safer to sharp-quote function names; use `#'`
- typst-preview.el#L379: Only use back/front quotes to link to top-level elisp symbols
- typst-preview.el#L379: Only use back/front quotes to link to top-level elisp symbols
- typst-preview.el#L392: Only use back/front quotes to link to top-level elisp symbols
- typst-preview.el#L392: Only use back/front quotes to link to top-level elisp symbols
- typst-preview.el#L411: Only use back/front quotes to link to top-level elisp symbols
- typst-preview.el#L411: Only use back/front quotes to link to top-level elisp symbols
- typst-preview.el#L425: Only use back/front quotes to link to top-level elisp symbols
- typst-preview.el#L425: Only use back/front quotes to link to top-level elisp symbols
- typst-preview.el#L437: Only use back/front quotes to link to top-level elisp symbols
- typst-preview.el#L470: Only use back/front quotes to link to top-level elisp symbols
- typst-preview.el#L470: Only use back/front quotes to link to top-level elisp symbols
- typst-preview.el#L470: Only use back/front quotes to link to top-level elisp symbols
```

⸺ Package and license:
- Use `:fetcher github :repo <repo>` instead of `:url`

riscy commented:
> I mistakenly thought shorthand prefixes like this were ok.

Thank you hugely for accommodating!  We make very occasional exceptions for them, but as you can see above, the available tooling generally isn't ready for this yet.  There's an open issue about it here: https://github.com/melpa/melpa/issues/9554#issuecomment-3165304380

Here's the latest -

⸺ `typst-preview.el` with byte-compile using Emacs 30.1:
```
typst-preview.el:64:11: Error: Cannot open load file: No such file or directory, f
```
I think a `Package-Requires` on f is the only remaining issue.

`#'load`-check on each file:
```
Loading typst-preview.el
  typst-preview.el:Error: Emacs 30.1:
  (file-missing "Cannot open load file" "No such file or directory" "f")
```

havarddj commented:
Thanks a lot, I totally understand!
Fixed the error now. (I removed the dependency on the f package, but forgot to remove the "require" expression! Classic.)

riscy commented:
Thanks again.  In case you haven't, ensure GitHub release v1.0.0-alpha.2 is up-to-date with your current code and `Package-Version`.  I'll merge now - and welcome to MELPA :)

havarddj commented:
Good call, I'll update the git and package versions. Thanks a lot for the time and effort!
